cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -1397,3 +1469,29 @@ func CleanSriov() {
 	}
 	sriov.WaitStable(sriovclient)
 }
+
+func getDeviceMac(pod *corev1.Pod, device string) (string, error) {
+	getMacOutput, err := pods.ExecCommand(client.Client, *pod, []string{"ip", "link", "show", "dev", device})
+	if err != nil {
+		return "", err
+	}
+	macLine := strings.Trim(strings.Split(getMacOutput.String(), "\n")[1], " ")
+	return strings.Split(strings.Trim(macLine, " "), " ")[1], nil
+}
+
+func getDeviceRXBytes(pod *corev1.Pod, device string) (int, error) {
+	statsCommand := []string{"ip", "-s", "l", "show", "dev", device}
+	stats, err := pods.ExecCommand(client.Client, *pod, statsCommand)
+	if err != nil {
+		return 0, err
+	}
+	statsLines := strings.Split(stats.String(), "\n")
+	for i, line := range statsLines {
+		if strings.Contains(strings.Trim(line, " "), "RX:") {
+			nextLine := strings.Trim(statsLines[i+1], " ")
```

Comment:
please check the len here before you do the `[i+1]` to be sure we are not going to fly